synapse/module_api/__init__.py

```diff
@@ -880,6 +880,16 @@ async def update_room_membership(
             content=content,
         )
 
+        # Update the m direct list
+        is_direct = content.get("is_direct", None)
```

Comment:
Is `is_direct` being in the content of an event something you've seen elsewhere? Looking at join events for my DMs I can't see an example of this.
Oh however I do see it in the *invite* event's content.
The usual way this seems to work (according to the spec) is that the client adds it to `m.direct` after accepting the invite (it could prompt the user first or ignore it, but Element doesn't do either of those).

After asking around, I think it would be better to expose a separate function for setting whether the room is in `m.direct` for a given user, and have the auto-accept module call into that (since it knows if an invite has `is_direct` set).

Reply:
Ah yes sorry I probably forgot to add the content part being passed to this pull request.

Reply:
I'm not aware of the per room one and what it's used for (or could be used for) so I'm guessing the global one is what we care about.

Reply:
Can I ask why it's better to separate this into two separate functions? 🙈

Reply:
I think there are a few reasons why a separate function arguably makes sense.

Clients have to make a separate request for setting a room as direct or not, so it makes some sense to reflect that design in the module API. Exposing account data is a bit more general than just the `m.direct` setting and may well have other uses in the future (for example, tags other than `m.direct`, lists of ignored users, and so on).
The two operations are also somewhat distinct; joining a room modifies room state, but setting it as direct modifies the *user's* account state. The current design doesn't leave a way for the status to be changed after the room is joined (and it also has no meaning if the target membership is anything other than 'join').